src/cloud-api-adaptor/hack/verify-provenance.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# Verify Github's attestation reports. Meant to verify binaries built
+# by upstream projects (kata-containers and guest-components).
+#
+# GH cli is used to verify.
+#
+# Asserts on the claims are:
+# - Triggered by push on the main branch
+# - Built on the given repository
+# - The gh action workflow is matching the given digest
+# - The code is matching the given digest
+#
+# -g will fetch attestation via gh cli, this requires GH_TOKEN to be
+# set. By default the attestation will be retrieved by walking the OCI
+# manifest
+
+set -euo pipefail
+
+usage() {
+	echo "Usage: $0 -a <oci-artifact> -d <expected-digest> -r <repository> [-g]"
+	exit 1
+}
+
+oci_artifact=""
+expected_digest=""
+repository=""
+github="0"
+
+# Parse options using getopts
+while getopts ":a:d:r:g" opt; do
+	case "${opt}" in
+	a)
+		oci_artifact="${OPTARG}"
+		;;
+	d)
+		expected_digest="${OPTARG}"
+		;;
+	r)
+		repository="${OPTARG}"
+		;;
+	g)
+		github="1"
+		;;
+	*)
+		usage
+		;;
+	esac
+done
+
+# Check if all required arguments are provided
+if [ -z "${oci_artifact}" ] || [ -z "${expected_digest}" ] || [ -z "${repository}" ]; then
+	usage
+fi
+
+if [[ "$oci_artifact" =~ @sha256:[a-fA-F0-9]{32}$ ]]; then
+	echo "The OCI artifact should be specified using its digest: my-repo.io/my-image@sha256:abc..."
+	exit 1
+fi
+
+# Convention by gh cli
+attestation_bundle="${oci_artifact#*@}.jsonl"
```

Comment:
I'm trying to decide if there is merit to clean this up at the end of the run, but I'm not sure it's worth it as we are likely to be using this inside the builder container.

Reply:
ah, forgot to address that. yeah, we can clean this up. i'll do it another pr.